### PR TITLE
Splitting code into modules

### PR DESCRIPTION
**Pull request by douglasmarques**
Modularising our ProseMirror implementation similar to the web version. 
This enhancement will result in having one JAR file per module, eliminating the requirement for clients to include the entire ProseMirror codebase in their applications - they will only add what is necessary. 
Additionally, this modification will make available the test-builder module that was previously not accessible since it was within a test folder.

### CODE REVIEW

File: history/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+              distribution.set("repo")
+            }
+          }
+        }
+      }
+    }
+
+    repositories {
+      maven {
+        url = uri("https://packages.atlassian.com/maven-central")
```
Comment (tchernykh) [inline history/build.gradle.kts:91]: Do we need this for Prosemirror?

Comment (douglasmarques) [reply history/build.gradle.kts:91]: This is where the jars are uploaded. Not sure if I understood the question

**Review by tchernykh — APPROVED** [review]

File: history/build.gradle.kts (hunk 1 above)
Comment (tchernykh) [reply history/build.gradle.kts:91]: ok, I thought we load something from it here.

File: REFERENCE_ATTRIBUTIONS.md [hunk 2]
```diff
@@ -131,3 +131,25 @@ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
+
+Prosemirror test builder](https://github.com/ProseMirror/prosemirror-test-builder)
```
Comment (encryptededdy) [inline REFERENCE_ATTRIBUTIONS.md:135]: Missing a `[` before Prosemirror test builder?

**Review by encryptededdy — APPROVED** [review]
Small typo

**Pull request merged**